xdsl/dialects/vector.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+from typing import Annotated, List
+
+from xdsl.dialects.builtin import IndexType, VectorType
+from xdsl.dialects.memref import MemRefType
+from xdsl.ir import Operation, SSAValue, Dialect, OpResult
+from xdsl.irdl import irdl_op_definition, Operand, VarOperand
+
+
+@irdl_op_definition
+class Load(Operation):
+    name = "vector.load"
+    memref: Annotated[Operand, MemRefType]
+    indices: Annotated[VarOperand, IndexType]
+    res: Annotated[OpResult, VectorType]
+
+    def verify_(self):
+        if self.memref.typ.element_type != self.res.typ.element_type:
```

Comment:
Could you test those lines (and the following ones, relative to verifier errors)?
You can do this directly using lit and filecheck, rather than in Python. You have an example in `arith_invalid.xdsl`.
I guess the best is to create a new `dialect/vector/` folder in `tests/filecheck/`.

Reply:
This sounds good!
I had one question though, since the program execution stops each time an exception is caught, shall I create a separate  `*vector_invalid.xdsl` file for each supposed exception? (a single file probably won't work because the execution will stop whenever the first exception is encountered)

Or is there a separate `filecheck` flag to handle such situations?

Reply:
This sounds good!
I had one question though, since the program execution stops each time an exception is caught, shall I create a separate  `*vector_invalid.xdsl` file for each supposed exception? (This sounds inelegant but a single file probably won't work because the execution will stop whenever the first exception is encountered)

Or is there a separate `filecheck` flag to handle such situations in the `filecheck` python package?
I tried finding something similar to [this](https://github.com/llvm/llvm-project/blob/e31d27e46048ccc3294d6b215dc778b3390e7834/mlir/test/Dialect/Arith/invalid.mlir#L4) directive, but couldn't find anything. The `filecheck` [roadmap ](https://github.com/mull-project/FileCheck.py/blob/main/FileCheck.pdf) doesn't seem to be providing anything useful either. 

Perhaps, the python package might implement a counterpart to `// expected-error @+1 {{}}` in future.

Reply:
Maybe a pytest makes more sense then? Or is there a reason why you suggested lit, @math-fehr?

Reply:
For now, I would say that the best would be to have a file per error, but we would put them all in a folder to make this clean.
I think we should clearly have an `expected-error`, it's just that its implementation is a bit of a pain a think, and that we didn't need it that much before.
And the reason why I suggested `lit` is that I think we should always prioritize `lit` over `pytest`. I think that `lit` is a far better abstraction than `pytest` for these kind of tests, that do not rely on custom `Python` code.

Reply:
Just to confirm, shall I create another PR to add `lit` tests for `vector.load/store` verifiers? 
This would contain 4 different files inside `tests/filecheck/dialects/vector` for each possible verifier exception.

Reply:
I would say to me that `lit` makes more sense, though I'm not sure that everyone agrees!